zetaclient/chains/bitcoin/observer/event.go

```diff
@@ -0,0 +1,233 @@
+	"github.com/zeta-chain/node/zetaclient/compliance"
+	"github.com/zeta-chain/node/zetaclient/config"
+	"github.com/zeta-chain/node/zetaclient/logs"
+)
+
+// InboundProcessability is an enum representing the processability of an inbound
+type InboundProcessability int
+
+const (
+	// InboundProcessabilityGood represents a processable inbound
+	InboundProcessabilityGood InboundProcessability = iota
+
+	// InboundProcessabilityDonation represents a donation inbound
+	InboundProcessabilityDonation
+
+	// InboundProcessabilityComplianceViolation represents a compliance violation
+	InboundProcessabilityComplianceViolation
+)
+
+// BTCInboundEvent represents an incoming transaction event
+type BTCInboundEvent struct {
+	// FromAddress is the first input address
+	FromAddress string
+
+	// ToAddress is the ZEVM receiver address
+	ToAddress string
+
+	// Value is the amount of BTC
+	Value float64
+
+	// DepositorFee is the deposit fee
+	DepositorFee float64
+
+	// MemoBytes is the memo of inbound
+	MemoBytes []byte
+
+	// MemoStd is the standard inbound memo if it can be decoded
+	MemoStd *memo.InboundMemo
+
+	// BlockNumber is the block number of the inbound
+	BlockNumber uint64
+
+	// TxHash is the hash of the inbound
+	TxHash string
+}
+
+// IsProcessable checks if the inbound event is processable
+func (event *BTCInboundEvent) CheckProcessability() InboundProcessability {
+	// compliance check on sender and receiver addresses
+	if config.ContainRestrictedAddress(event.FromAddress, event.ToAddress) {
+		return InboundProcessabilityComplianceViolation
+	}
+
+	// compliance check on receiver, revert/abort addresses in standard memo
+	if event.MemoStd != nil {
+		if config.ContainRestrictedAddress(
+			event.MemoStd.Receiver.Hex(),
+			event.MemoStd.RevertOptions.RevertAddress,
+			event.MemoStd.RevertOptions.AbortAddress,
+		) {
+			return InboundProcessabilityComplianceViolation
+		}
+	}
+
+	// donation check
+	if bytes.Equal(event.MemoBytes, []byte(constant.DonationMessage)) {
+		return InboundProcessabilityDonation
+	}
+
+	return InboundProcessabilityGood
+}
+
+// DecodeEventMemoBytes decodes the memo bytes as either standard or legacy memo
+func (ob *Observer) DecodeEventMemoBytes(event *BTCInboundEvent) error {
+	var (
+		err      error
+		memoStd  *memo.InboundMemo
+		receiver ethcommon.Address
+	)
+
+	// skip decoding donation tx as it won't go through zetacore
+	if bytes.Equal(event.MemoBytes, []byte(constant.DonationMessage)) {
+		return nil
+	}
+
+	// try to decode the standard memo as the preferred format
+	// the standard memo is NOT enabled for Bitcoin mainnet
+	if ob.Chain().ChainId != chains.BitcoinMainnet.ChainId {
+		memoStd, err = memo.DecodeFromBytes(event.MemoBytes)
+	}
+
+	switch {
+	case memoStd != nil:
+		// skip memo that carries improper data
+		if err != nil {
+			return errors.Wrap(err, "standard memo contains improper data")
+		}
+
+		// NoAssetCall will be disabled for Bitcoin until full V2 support
```

Comment:
```suggestion
		// NoAssetCall will be disabled for Bitcoin until full V2 support
		// https://github.com/zeta-chain/node/issues/2711
```

Reply:
updated in https://github.com/zeta-chain/node/pull/3025/commits/7a2880bf21e47dfa10d0f3a9dd3756064237afbd